frame/support/src/traits/tokens/fungible.rs

```diff
@@ -32,41 +32,65 @@ pub use imbalance::{Imbalance, HandleImbalanceDrop, DebtOf, CreditOf};
 pub trait Inspect<AccountId> {
 	/// Scalar type for representing balance of an account.
 	type Balance: Balance;
+
 	/// The total amount of issuance in the system.
 	fn total_issuance() -> Self::Balance;
+
 	/// The minimum balance any single account may have.
 	fn minimum_balance() -> Self::Balance;
+
 	/// Get the balance of `who`.
 	fn balance(who: &AccountId) -> Self::Balance;
+
+	/// Get the maximum amount that `who` can withdraw/transfer successfully.
+	fn withdrawable_balance(who: &AccountId) -> Self::Balance { Self::balance(who) }
+
 	/// Returns `true` if the balance of `who` may be increased by `amount`.
 	fn can_deposit(who: &AccountId, amount: Self::Balance) -> DepositConsequence;
+
 	/// Returns `Failed` if the balance of `who` may not be decreased by `amount`, otherwise
 	/// the consequence.
 	fn can_withdraw(who: &AccountId, amount: Self::Balance) -> WithdrawConsequence<Self::Balance>;
```

Comment:
```suggestion
	fn can_burn(who: &AccountId, amount: Self::Balance) -> WithdrawConsequence<Self::Balance>;
```

Reply:
it would be `can_burn_from`, but that's quite dumb - since it's the `from` part that's important here, not the `burn`. `can_decrease_balance` would be the obvious name, but it's needlessly long.